### PR TITLE
Sonar Cloud Workflow

### PR DESCRIPTION
**Pull request by juliojimenez**
Signed-off-by: Julio Jimenez <julio@clickhouse.com>

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  sonar_cloud:
+    name: Sonar Cloud
+    runs-on: [self-hosted, i-dont-know]
```
Comment (juliojimenez) [inline .github/workflows/sonar.yml:11]: @Felixoid which self-hosted runner should I use here?

Comment (juliojimenez) [reply .github/workflows/sonar.yml]: I picked `builder` 🤷‍♂️

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    runs-on: [self-hosted, builder]
+    env:
+      SONAR_SCANNER_VERSION: 4.7.0.2747
+      SONAR_SERVER_URL: "https://sonarcloud.io"
+      BUILD_WRAPPER_OUT_DIR: build_wrapper_output_directory # Directory where build-wrapper output will be placed
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
```
Comment (Felixoid) [inline .github/workflows/sonar.yml:23]: It looks like a good candidate for Dockerfile to me

File: .github/workflows/sonar.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+name: Sonar Cloud
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (Felixoid) [inline .github/workflows/sonar.yml:7]: Is it just now? Will we run it nightly in the end?

File: .github/workflows/sonar.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
+          echo "$HOME/.sonar/build-wrapper-linux-x86" >> $GITHUB_PATH
+      - name: Set Up Build Tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install -yq git cmake ccache python3 ninja-build
+          sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
+      - name: Run build-wrapper
+        run: |
+          mkdir build
```
Comment (Felixoid) [inline .github/workflows/sonar.yml:49]: It looks like the same workflow is used in Coverity. https://github.com/ClickHouse/ClickHouse/blob/master/.github/workflows/nightly.yml#L82-L109

So the image for it should be created on top of https://github.com/ClickHouse/ClickHouse/tree/master/docker/packager/binary

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (Felixoid) [reply .github/workflows/sonar.yml]: the builder looks like a proper choice!

**Comment by evillique** [comment]
@juliojimenez
Style check fails in the result of these changes: [report](https://s3.amazonaws.com/clickhouse-test-reports/0/acc03cf52e7079110cc99f2c17854672697e4f80/style_check.html), and since style check is required for other tests to run, CI in all PRs fail.

**Comment by alexey-milovidov** [comment]
Style check is broken, reverting...

```
/ClickHouse/.github/workflows/sonar.yml
  2:1       warning  truthy value should be one of [false, true]  (truthy)
```

**Comment by alexey-milovidov** [comment]
@juliojimenez 

> Not for changelog (changelog entry is not required)

This is wrong, should be "Build/Testing/Packaging Improvement".

**Pull request merged**